nrf52-hal-common/src/rng.rs

```diff
@@ -4,6 +4,7 @@
 
 
 use core::ops::Deref;
+use core::sync::atomic::{compiler_fence, Ordering::AcqRel};
```

Comment:
How deliberate is your choice of `AcqRel`? `SeqCst` seems more strict and easier to understand, so it would be my choice. Not saying `AcqRel` is wrong, just saying I'm not 100% sure I understand what it's doing (and if that's what we need), while `SeqCst` seems quite straight-forward.

Reply:
Moved to SeqCst, we can re-evaluate later if necessary